packages/swapper/src/swappers/zrx/zrxApproveInfinite/zrxApproveInfinite.ts

```diff
@@ -1,37 +1,67 @@
-import { ApproveInfiniteInput, EvmSupportedChainIds, SwapError, SwapErrorTypes } from '../../../api'
+import {
+  ApproveAmountInput,
+  ApproveInfiniteInput,
+  EvmSupportedChainIds,
+  SwapError,
+  SwapErrorTypes
+} from '../../../api'
+import { grantAllowance } from '../../../swappers/utils/helpers/helpers'
 import { erc20Abi } from '../../utils/abi/erc20-abi'
 import { APPROVAL_GAS_LIMIT } from '../../utils/constants'
-import { grantAllowance } from '../../utils/helpers/helpers'
 import { ZrxSwapperDeps } from '../types'
 import { MAX_ALLOWANCE } from '../utils/constants'
 
-export async function zrxApproveInfinite<T extends EvmSupportedChainIds>(
+const grantAllowanceForAmount = async <T extends EvmSupportedChainIds>(
   { adapter, web3 }: ZrxSwapperDeps,
-  { quote, wallet }: ApproveInfiniteInput<T>
-) {
-  try {
-    const approvalQuote = {
-      ...quote,
-      sellAmount: MAX_ALLOWANCE,
-      feeData: {
-        ...quote.feeData,
-        chainSpecific: {
-          ...quote.feeData.chainSpecific,
-          // 0x approvals are cheaper than trades, but we don't have dynamic quote data for them.
-          // Instead, we use a hardcoded gasLimit estimate in place of the estimatedGas in the 0x quote response.
-          estimatedGas: APPROVAL_GAS_LIMIT
-        }
+  { quote, wallet }: ApproveInfiniteInput<T>,
+  approvalAmount: string
+) => {
+  const approvalQuote = {
+    ...quote,
+    sellAmount: approvalAmount,
+    feeData: {
+      ...quote.feeData,
+      chainSpecific: {
+        ...quote.feeData.chainSpecific,
+        // 0x approvals are cheaper than trades, but we don't have dynamic quote data for them.
+        // Instead, we use a hardcoded gasLimit estimate in place of the estimatedGas in the 0x quote response.
+        estimatedGas: APPROVAL_GAS_LIMIT
       }
     }
-    const allowanceGrantRequired = await grantAllowance<T>({
-      quote: approvalQuote,
-      wallet,
-      adapter,
-      erc20Abi,
-      web3
+  }
+  return await grantAllowance<T>({
+    quote: approvalQuote,
+    wallet,
+    adapter,
+    erc20Abi,
+    web3
+  })
+}
+
+export async function zrxApproveAmount<T extends EvmSupportedChainIds>(
```

Comment:
This module is named `zrxApproveInfinite`, it should either be renamed, or this should be in a whole new module (preference for the former vs. having to search through many different modules, this should be colocated IMO)

Reply:
or even better, wax all of these nested directories and just have the swapper class with the logic contained within there :) not a fan of this bloated setup personally. obv not for this PR.

I would go with rename as well if we want to change at this time

Reply:
Agree, and so how would we call the module containing these 2 functions? `srxApprove`?

Reply:
Precisely @NeOMakinG, that's what I had in mind